File: app/src/main/java/com/example/core/ai/AiTelemetry.kt
package com.example.core.ai

import android.util.Log

/**
 * Privacy-safe debug telemetry for on-device AI paths. Logs inference route + latency only —
 * never prompt or response text.
 */
object AiTelemetry {
    private const val TAG = "MeridianAI"

    fun logInference(path: String, provenance: AiProvenance, latencyMs: Long) {
        Log.d(TAG, "path=$path provenance=$provenance latencyMs=$latencyMs")
    }
}

File: ios/MeridianApp/Core/AI/AiTelemetry.swift
// AiTelemetry.swift
// Meridian — iOS 27 / Swift 6
//
// Privacy-safe debug telemetry — logs inference path + latency only.

import Foundation
import os

enum AiTelemetry {
    private static let log = Logger(subsystem: "com.meridian.app", category: "AI")

    static func logInference(path: String, source: InferenceSource, latencyMs: Int64) {
        log.debug("path=\(path, privacy: .public) source=\(source.rawValue, privacy: .public) latencyMs=\(latencyMs)")
    }
}
